ext/bcmath/libbcmath/src/raise.c

```diff
@@ -52,12 +52,16 @@ bc_raise (bc_num num1, bc_num num2, bc_num *result, int scale)
    int calcscale;
    char neg;
 
-   /* Check the exponent for scale digits and convert to a long. */
-   if (num2->n_scale != 0)
-     php_error_docref (NULL, E_WARNING, "Non-zero scale in exponent");
-   exponent = bc_num2long (num2);
-   if (exponent == 0 && (num2->n_len > 1 || num2->n_value[0] != 0))
-       php_error_docref (NULL, E_WARNING, "Exponent too large");
+	/* Check the exponent for scale digits and convert to a long. */
+	if (num2->n_scale != 0) {
+		/* 2nd argument from PHP_FUNCTION(bcpow) */
+		zend_argument_value_error(2, "must be an integer");
```

Comment:
This is missing a return. I also think that "must be an integer" is a bit confusing here, in that it sounds like "must be of type int". Maybe keep "must have zero scale" or use "cannot have a fractional part".